tensorboard/plugins/graph/tf_graph_common/op.ts

```diff
@@ -343,6 +343,14 @@ module tf.graph.op {
    * Tensorflow operation that is valid for the TPU.
    */
   export function opValid(opNode: OpNode): boolean {
+    // Function library nodes are generally for internal use.
+    if (opNode.name.search(FUNCTION_LIBRARY_NODE_PREFIX) == 0) {
```

Comment:
Functions in general can run on TPUs? OK

Reply:
Function libraries may or may not run on TPUs depending on the ops within them.  

I am choosing to omit function library checking, because some of our APIs (e.g. Datasets) use them to hold non-TPU ops that will end up being placed on the CPU.  A common example of these are input pipeline ops in Datasets.

This compatibility check is not meant to be exact right now, so we err on the side of encouraging the user to try compiling their graph.  I would like to put some kind of caution statement somewhere in the UI..

Reply:
I see. Thanks for the clarification. I wonder what we can do for the UI ... maybe introduce a yellow color, and then more info is revealed in the info card upon node selection.